### DISCUSSION TITLE
Issue with long PDF + pagination

### DISCUSSION THREAD

**Discussion by ghost** [hrynko/vue-pdf-embed#166] [Q&A] [answered]
Hello,

I have a problem where I want to display a long PDF (over 6000 pages), but only blank pages are being rendered. If I filter by single page, it works fine.

I wanted to integrate pagination, but I can't find a way to display a range of pages, for example, pages 1 to 50. How can I do this?

thanks,

Regards

**Comment by hrynko** [comment, marked as answer]
Hi @antoinedu06160,

Indeed, I would recommend displaying large documents paginated (page by page or within ranges). At some point some kind of lazy loading could be added to the lib, but for now you could use it as follows:

```vue
<template>
  <div v-if="pdfSource">
    <vue-pdf-embed
      v-for="page in pages"
      :key="page"
      :source="pdfSource"
      :page="page"
    />
  </div>
</template>

<script>
import VuePdfEmbed from 'vue-pdf-embed'

export default {
  components: {
    VuePdfEmbed,
  },
  data() {
    return {
      pages: [...Array(51).keys()].slice(1),
      pdfSource: null,
    }
  },
  async mounted() {
    this.pdfSource = await VuePdfEmbed.getDocument('<PDF_URL>').promise
  },
}
</script>
```
